src/sage/combinat/bijectionist.py

```diff
@@ -0,0 +1,3230 @@
+        sage: bij.set_constant_blocks(orbit_decomposition(A, theta))
+        sage: list(bij.solutions_iterator())
+        []
+
+    An example identifying `s` and `S`::
+
+        sage: N = 4
+        sage: A = [dyck_word for n in range(1, N) for dyck_word in DyckWords(n)]
+        sage: B = [binary_tree for n in range(1, N) for binary_tree in BinaryTrees(n)]
+        sage: concat_path = lambda D1, D2: DyckWord(list(D1) + list(D2))
+        sage: concat_tree = lambda B1, B2: concat_path(B1.to_dyck_word(),
+        ....:                                          B2.to_dyck_word()).to_binary_tree()
+        sage: bij = Bijectionist(A, B)
+        sage: bij.set_intertwining_relations((2, concat_path, concat_tree))
+        sage: bij.set_statistics((lambda d: d.semilength(), lambda t: t.node_number()))
+        sage: for D in sorted(bij.minimal_subdistributions_iterator(), key=lambda x: (len(x[0][0]), x)):
+        ....:     ascii_art(D)
+        ( [ /\ ], [ o ] )
+        (           [ o   ] )
+        (           [  \  ] )
+        ( [ /\/\ ], [   o ] )
+        (           [   o ] )
+        ( [  /\  ]  [  /  ] )
+        ( [ /  \ ], [ o   ] )
+        (             [ o     ] )
+        (             [  \    ] )
+        (             [   o   ] )
+        (             [    \  ] )
+        ( [ /\/\/\ ], [     o ] )
+        (             [ o   ] )
+        (             [  \  ] )
+        (             [   o ] )
+        ( [    /\  ]  [  /  ] )
+        ( [ /\/  \ ], [ o   ] )
+        (             [   o   ] )
+        ( [  /\    ]  [  / \  ] )
+        ( [ /  \/\ ], [ o   o ] )
+        (                     [   o,     o ] )
+        (                     [  /      /  ] )
+        ( [           /\   ]  [ o      o   ] )
+        ( [  /\/\    /  \  ]  [  \    /    ] )
+        ( [ /    \, /    \ ], [   o  o     ] )
+
+    The output is in a form suitable for FindStat::
+
+        sage: findmap(list(bij.minimal_subdistributions_iterator()))            # optional -- internet
+        0: Mp00034 (quality [100])
+        1: Mp00061oMp00023 (quality [100])
+        2: Mp00018oMp00140 (quality [100])
+
+    TESTS::
+
+        sage: N = 4; A = B = [permutation for n in range(N) for permutation in Permutations(n)]
+        sage: theta = lambda pi: Permutation([x+1 if x != len(pi) else 1 for x in pi[-1:]+pi[:-1]])
+        sage: def tau(pi):
+        ....:    n = len(pi)
+        ....:    return sum([1 for i in range(1, n+1) for j in range(1, n+1)
+        ....:                if i<j <= pi(i)<pi(j) or pi(i)<pi(j)<i<j])
+        sage: bij = Bijectionist(A, B, tau)
+        sage: bij.set_statistics((len, len))
+        sage: from sage.combinat.cyclic_sieving_phenomenon import orbit_decomposition
+        sage: bij.set_constant_blocks(orbit_decomposition(A, theta))
+        sage: for solution in sorted(list(bij.solutions_iterator()), key=lambda d: tuple(sorted(d.items()))):
```

Comment:
```suggestion
        sage: for solution in sorted(bij.solutions_iterator(), key=lambda d: sorted(d.items())):
```
The result of `sorted` is a list and compares the same as a tuple.